pandas/core/generic.py

```diff
@@ -10703,8 +10705,24 @@ def rolling(
             axis=0,
             closed=None,
         ):
+            if not isinstance(self, (ABCSeries, ABCDataFrame)):
```

Comment:
why do we need this check? would prefer not to see ABCSeries and ABCDataFrame in core.generic.

Reply:
yeah this should be *inside* Rolling constructor. Let's try to keep this diff as copy-paste as possible though & I don't think this was on the original diff at all?

Reply:
This was in the original "top level" `rolling` function that would be imported here, but I found a better place to put this check.